spec/jobs/operation_job_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe OperationJob do
+  let(:operation) do
+    create(:operation,
+      job_class: subject,
+      job_id: nil,
+      job_started_at: nil,
+      job_completed_at: nil)
+  end
+
+  describe "#perform" do
+    subject do
+      Class.new(described_class) do
+        def perform(operation, *args)
+        end
+      end
+    end
```

Comment:
Slick! But I don't think this is your subject. The operation object is the subject I think because it's the thing you're `expect`ing, right?

Reply:
Also please always use named subjects. So `subject(:operation) { ... }`

Reply:
The subject under test in this spec file is the `OperationJob` class so I thought it made sense for the subclass implementation to be the subject.  There are expectations about the `operation` but only to test behavior in the `OperationJob` class.

Reply:
Ok sounds good to me!